numbat/src/tokenizer.rs

```diff
@@ -539,6 +539,37 @@ impl Tokenizer {
                     });
                 }
             },
+            ':' if self.interpolation_state.is_inside() => {
+                while self
+                    .peek()
+                    .map(|c| c != '"' && c != ':' && c != '}')
+                    .unwrap_or(false)
+                {
+                    self.advance();
+                }
+
+                if self.peek() == Some('"') {
+                    return Err(TokenizerError {
+                        kind: TokenizerErrorKind::UnterminatedStringInterpolation,
+                        span: Span {
+                            start: self.token_start,
+                            end: self.current,
+                            code_source_id: self.code_source_id,
+                        },
+                    });
+                } else if self.peek() == Some('}') || self.peek() == Some(':') {
+                    TokenKind::StringInterpolationSpecifier
+                } else {
+                    return Err(TokenizerError {
+                        kind: TokenizerErrorKind::UnterminatedString,
+                        span: Span {
+                            start: self.token_start,
+                            end: self.current,
+                            code_source_id: self.code_source_id,
+                        },
+                    });
+                }
+            }
```

Comment:
The lexer does little more than just have the whole specifier as a token, would be nice to make the tokens more specific. For instance, if we add support for `:0.3f`, maybe it should be tokenized to "colon, specifer(0.3f)" or maybe "colon, number(0.3), text("f")` or something like that? idk

Also should we allow whitespace between the specifier parts?

Reply:
I think we decided against this (for now)?

Reply:
Yeah, we're just passing it all to `strfmt` anyway.